src/core/lib/security/transport/secure_endpoint.cc

```diff
@@ -143,12 +163,51 @@ static void secure_endpoint_unref(secure_endpoint* ep) {
 static void secure_endpoint_ref(secure_endpoint* ep) { gpr_ref(&ep->ref); }
 #endif
 
+static void post_reclaimer(secure_endpoint* ep) {
+  if (!ep->has_posted_reclaimer) {
+    SECURE_ENDPOINT_REF(ep, "benign_reclaimer");
+    ep->has_posted_reclaimer = true;
+    ep->memory_owner.PostReclaimer(
+        grpc_core::ReclamationPass::kBenign,
+        [ep](absl::optional<grpc_core::ReclamationSweep> sweep) {
+          if (sweep.has_value()) {
+            if (GRPC_TRACE_FLAG_ENABLED(grpc_resource_quota_trace)) {
+              gpr_log(GPR_INFO,
+                      "secure endpoint: benign reclamation to free memory");
+            }
+            grpc_slice temp_read_slice;
+            grpc_slice temp_write_slice;
+
+            ep->read_mu.Lock();
+            temp_read_slice = ep->read_staging_buffer;
+            ep->read_staging_buffer = grpc_empty_slice();
+            ep->read_mu.Unlock();
+
+            ep->write_mu.Lock();
+            temp_write_slice = ep->write_staging_buffer;
+            ep->write_staging_buffer = grpc_empty_slice();
+            ep->write_mu.Unlock();
+
+            grpc_slice_unref_internal(temp_read_slice);
+            grpc_slice_unref_internal(temp_write_slice);
+            ep->has_posted_reclaimer = false;
+          }
+          SECURE_ENDPOINT_UNREF(ep, "benign_reclaimer");
+        });
+  }
+}
+
 static void flush_read_staging_buffer(secure_endpoint* ep, uint8_t** cur,
```

Comment:
since this function gives out pointers to things protected by read_mu, the entire function should be protected by read_me

Reply:
Done